src/core/Microsoft.PowerToys.Settings.UI.UnitTests/Mocks/IIOProviderMocks.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.PowerToys.Settings.UI.Lib.Utilities;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerToys.Settings.UI.UnitTests.Mocks
+{
+    internal static class IIOProviderMocks
+    {
+
+        /// <summary>
+        /// This method mocks an IO provider to validate tests wich required saving to a file, and then reading the contents of that file, or verifying it exists
+        /// </summary>
+        /// <returns></returns>
+        internal static Mock<IIOProvider> GetMockIOProviderForSaveLoadExists()
+        {
+            string savePath = string.Empty;
+            string saveContent = string.Empty;
+            var mockIOProvider = new Mock<IIOProvider>();
+            mockIOProvider.Setup(x => x.WriteAllText(It.IsAny<string>(), It.IsAny<string>()))
```

Comment:
[Clarification] Will the savePath and saveContent variables be accessed by reference? I'm confused on why the mocked methods would work since savePath and saveContent would go out of reference at the end of this method and get garbage collected.

Reply:
savePath and contentPath, being strings, are reference types, and therefore they will be marked by the garbage collector and won't be collected when the collection gets triggered.  Unlike C++ which would clean the memory up when it goes out of scope(unless it was RVOd), the memory used by savePath and contentPath won't get freed until memory pressure intiates garbage collection.